### PR TITLE
REF: move MaskedArray subclass attributes to dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
Discussed on yesterday's dev call, incremental steps towards merging IntegerArray/FloatingArray/BooleanArray into a single class

### CODE REVIEW

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -1542,6 +1543,25 @@ class BaseMaskedDtype(ExtensionDtype):
 
     base = None
     type: type
+    _internal_fill_value: Scalar
+
+    @property
+    def truthy_value(self):
```
Comment (mroeschke) [inline pandas/core/dtypes/dtypes.py:1549]: Should these be "private" ie `_truthy_value?

Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py:1549]: they are accessed from outside the dtype's methods

Comment (WillAyd) [reply pandas/core/dtypes/dtypes.py:1549]: Shouldn't these raise NotImplementedError as a fallback? The idea is this class can be used for more than just float/integer/bool in the future right?

File: pandas/core/arrays/masked.py [hunk 2]
```diff
@@ -155,8 +148,9 @@ def _from_sequence(cls, scalars, *, dtype=None, copy: bool = False) -> Self:
     @classmethod
     @doc(ExtensionArray._empty)
     def _empty(cls, shape: Shape, dtype: ExtensionDtype) -> Self:
-        values = np.empty(shape, dtype=dtype.type)
-        values.fill(cls._internal_fill_value)
+        dtype = cast(BaseMaskedDtype, dtype)
+        values: np.ndarray = np.empty(shape, dtype=dtype.type)
```
Comment (WillAyd) [inline pandas/core/arrays/masked.py:152]: Is the type declaration required here? I assume mypy would be able to infer this from the right hand side

File: pandas/core/arrays/masked.py [hunk 3]
```diff
@@ -155,8 +148,9 @@ def _from_sequence(cls, scalars, *, dtype=None, copy: bool = False) -> Self:
     @classmethod
     @doc(ExtensionArray._empty)
     def _empty(cls, shape: Shape, dtype: ExtensionDtype) -> Self:
-        values = np.empty(shape, dtype=dtype.type)
-        values.fill(cls._internal_fill_value)
+        dtype = cast(BaseMaskedDtype, dtype)
```
Comment (WillAyd) [inline pandas/core/arrays/masked.py:151]: The cast suspicious - why wouldn't we just change the type of `dtype` to `BaseMaskedType` if the `_internal_fill_value` attribute was required?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/masked.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/masked.py:152]: mypy complained without this

File: pandas/core/dtypes/dtypes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py:1549]: i dont think there's any plans to extend like that

File: pandas/core/arrays/masked.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/arrays/masked.py:151]: IIRC mypy complains if we do that, but in practice that is correct that we should only ever get BaseMaskedDtype here

File: pandas/core/dtypes/dtypes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py]: On second thought decided to privatize after all

**Review by mroeschke — APPROVED** [review]
LGTM

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**